### PR TITLE
[@mantine/core] Add Cascader Component

### PR DESCRIPTION
**Pull request by tsar-boomba**
This is a pull request for a cascader component as requested in #757.
The main functionality of the component is present, but I will need to add more stories, demos, and tests, as well as more polish to appearance and functionality.

### CODE REVIEW

File: src/mantine-labs/src/Cascader/CascaderDropdown/CascaderDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  dropdownComponent?: React.FC<any>;
+  referenceElement?: HTMLElement;
+  direction?: React.CSSProperties['flexDirection'];
+  onDirectionChange?: (direction: React.CSSProperties['flexDirection']) => void;
+  switchDirectionOnFlip?: boolean;
+  zIndex?: number;
+  dropdownPosition?: 'bottom' | 'top' | 'flip';
+}
+
+export const CascaderDropdown = forwardRef<HTMLDivElement, CascaderDropdownProps>(
```
Comment (rtivital) [inline src/mantine-labs/src/Cascader/CascaderDropdown/CascaderDropdown.tsx:28]: Reuse SelectDropdown, no need to copy component

File: src/mantine-labs/src/Cascader/CascaderMenus/CascaderMenus.styles.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { createStyles, MantineSize } from '@mantine/core';
```
Comment (rtivital) [inline src/mantine-labs/src/Cascader/CascaderMenus/CascaderMenus.styles.ts:1]: If styles are not used, remove styles file

File: src/mantine-labs/src/Cascader/CascaderMenus/CascaderMenus.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
+import { ClassNames, DefaultProps, MantineSize } from '@mantine/core';
```
Comment (rtivital) [inline src/mantine-labs/src/Cascader/CascaderMenus/CascaderMenus.tsx:1]: Rename component to CascaderItems

File: src/mantine-labs/src/Cascader/CascaderRightSection/CascaderRightSection.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  shouldClear: boolean;
+  clearButtonLabel?: string;
+  onClear?: () => void;
+  size: MantineSize;
+  error?: any;
+  // eslint-disable-next-line react/no-unused-prop-types
+  disabled?: boolean;
+}
+
+export function CascaderRightSection({
```
Comment (rtivital) [inline src/mantine-labs/src/Cascader/CascaderRightSection/CascaderRightSection.tsx:16]: Reuse SelectRightSection

File: src/mantine-labs/src/Cascader/CascaderRightSection/ChevronIcon.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+const iconSizes = {
+  xs: 14,
+  sm: 18,
+  md: 20,
+  lg: 24,
+  xl: 28,
+};
+
+export function ChevronIcon({ size, error, style, ...others }: ChevronIconProps) {
```
Comment (rtivital) [inline src/mantine-labs/src/Cascader/CascaderRightSection/ChevronIcon.tsx:17]: Reuse Select ChevronIcon

File: src/mantine-labs/src/Cascader/Menu/Menu.tsx [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+  onItemHover: React.Dispatch<React.SetStateAction<number[]>>;
+  onItemSelect(item: CascaderItem, index: number): void;
+  size: MantineSize;
+  itemComponent: React.FC<any>;
+  menuComponent: React.FC<any>;
+  nesting: number;
+  maxDropdownHeight: number;
+}
+
+export const Menu = forwardRef<HTMLDivElement, CascaderMenuProps>(
```
Comment (rtivital) [inline src/mantine-labs/src/Cascader/Menu/Menu.tsx:25]: Rename component to CascaderItemsList

File: src/mantine-labs/src/Cascader/getItem.ts [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import { CascaderItem } from './types';
+
+export function getItem(
+  data: CascaderItem[],
+  nesting: number,
+  indexes: number[] | null
```
Comment (rtivital) [inline src/mantine-labs/src/Cascader/getItem.ts:6]: do not use `| null`, Mantine does not use strict mode

File: src/mantine-labs/src/Cascader/index .ts [hunk 8]
```diff
@@ -0,0 +1,4 @@
+export { Cascader } from './Cascader';
+
+export type { CascaderProps } from './Cascader';
```
Comment (rtivital) [inline src/mantine-labs/src/Cascader/index .ts:3]: export CascaderStylesNames

**Review by rtivital — COMMENTED** [review]
Initial review, I haven't yet reviewed how component works

**Review by rtivital — COMMENTED** [review]
Initial review, I haven't yet reviewed how component works

**Comment by lightsound** [comment]
@rtivital 
Do you plan to add Cascader?
Why did you close this PR?

**Pull request closed without merging**